### PR TITLE
[feat]: [CDS-75131] : adding ng setting as OR with CDS_PROJECT_SCOPED_RESOURCE_CONSTRAINT_QUEUE

### PR DESCRIPTION
**Pull request by tchaurasiya**
## Describe your changes

* Adding NG Setting "project_scoped_resource_constraint_queue" as or condition with FF 
CDS_PROJECT_SCOPED_RESOURCE_CONSTRAINT_QUEUE
* Further plan is to enable setting as ui available for customers then remove the FF 

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51212)

### CODE REVIEW

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/stage/DeploymentStagePMSPlanCreatorV2.java [hunk 1]
```diff
@@ -281,6 +289,15 @@ public LinkedHashMap<String, PlanCreationResponse> createPlanForChildrenNodes(
     }
   }
 
+  private boolean isProjectScopedResourceConstraintQueueByFFOrSetting(PlanCreationContext ctx) {
+    return featureFlagHelperService.isEnabled(
+               ctx.getAccountIdentifier(), FeatureName.CDS_PROJECT_SCOPED_RESOURCE_CONSTRAINT_QUEUE)
+        || parseBoolean(NGRestUtils
+                            .getResponse(settingsClient.getSetting(
+                                PROJECT_SCOPED_RESOURCE_CONSTRAINT_SETTING_ID, ctx.getAccountIdentifier(), null, null))
+                            .getValue());
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/stage/DeploymentStagePMSPlanCreatorV2.java:298]: dont we need to org/projectId here?

Comment (ivan-83) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/stage/DeploymentStagePMSPlanCreatorV2.java:298]: yes, the same question.

Comment (tchaurasiya) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/stage/DeploymentStagePMSPlanCreatorV2.java:298]: No, as its account level setting and cant be overridden at org/project level, we don't need it.


```
  - identifier: project_scoped_resource_constraint_queue
    name: Project Scoped Resource Constraint Queue
    category: CD
    defaultValue: false
    valueType: BOOLEAN
    allowOverrides: false
    allowedScopes:
      - ACCOUNT
```

**Review by lovish1999 — DISMISSED** [review]

**Review by ivan-83 — DISMISSED** [review]

**Comment by tchaurasiya** [comment]
trigger uts

**Comment by tchaurasiya** [comment]
trigger uts

**Review by ivan-83 — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51212#-:-NarT0Bn4pOt--ueDKPy:bnfp4nl)* status: 2 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by yogesh-chauhan — APPROVED** [review]

**Comment by tchaurasiya** [comment]
trigger review

**Comment by tchaurasiya** [comment]
trigger messagecheck

**Pull request merged**